card-game.js

```diff
@@ -0,0 +1,60 @@
+class Deck {
+    constructor(cards) {
+        this.cards = cards // list of Cards
+    }
+    
+    get_card(quantity) {
+        // return Card
+    }
+
+    shuffle(times) {
+        // return nothing
+    }
+
+    cut(index) {
+        // return nothing
+    }
+} 
+
+class Card {
+    constructor(suit, value, name) {
+        this.suit = suit; // Suit
+        this.value = value; // int
+        this.name = name; // string
+    }
+}
+
+class Suit {
+    constructor(name, color) {
+        this.name = name; // string
+        this.color = color; // string
+    }
+}
+
+class Player {
+    constructor(name, hand, score) {
+        this.name = name; // string
+        this.hand = hand; // Hand
+        this.score = score; // int
+    }
+
+    change_score(score) -> this.score
+}
+
+class Game {
+    constructor(name, deck, players) {
```

Comment:
`name` and `deck` are intrinsic concepts of the game.  Whatever classes extend this base class should define those properties and not expose them through their own constructors.